apps/web/pages/api/auth/[...nextauth].ts

```diff
@@ -69,23 +75,49 @@ export default NextAuth({
         };
       },
     }),
+    GoogleProvider({
+      clientId: process.env.GOOGLE_CLIENT_ID!,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
+    }),
   ],
   callbacks: {
     async jwt({ token, user, account }) {
       return {
         ...token,
       };
     },
-    async session({ session, token }) {
+    async session({ session, token, user }) {
       const documensoSession: Session = {
         ...session,
         user: {
           ...session.user,
+          id: Number(token.sub),
         },
       };
 
       documensoSession.expires;
       return documensoSession;
     },
+    async signIn({ user, account }) {
+      if (account?.provider === "google") {
+        await prisma.user
+          .update({
+            where: {
+              email: user.email!,
+            },
+            data: {
+              identityProvider: IdentityProvider.GOOGLE,
+            },
+          })
+          .then(() => {
+            return true;
+          });
+      }
+
+      return true;
+    },
```

Comment:
Should work just fine, no need to mix .then/.catch with async/await

```suggestion
    async signIn({ user, account }) {
      if (account?.provider === "google") {
        await prisma.user
          .update({
            where: {
              email: user.email!,
            },
            data: {
              identityProvider: IdentityProvider.GOOGLE,
            },
          })
      }

      return true;
    },
```

Reply:
The auth was failing before because the user is being updated before it is created in the `signIn` callback.

Initially, it may have appeared to work because I had a user with the same email already present in the local db. This situation might have been the case for @Mythie as well. 

However, when starting with a fresh database that only includes the seed user, this callback causes the authentication process to fail.